File: stacks/backend/PHP/src/Config/ChainRiceConfig.php
<?php

namespace ChainRice\Bridge\Config;

/**
 * Configuration class for ChainRice client.
 */
class ChainRiceConfig
{
    private string $accountingApiUrl = 'http://localhost:8080';
    private string $taxApiUrl = 'http://localhost:8081';
    private string $blockchainUrl = 'http://localhost:26657';
    private string $apiKey = '';
    private int $timeout = 30;

    public function __construct(
        ?string $accountingApiUrl = null,
        ?string $taxApiUrl = null,
        ?string $blockchainUrl = null,
        ?string $apiKey = null,
        ?int $timeout = null
    ) {
        if ($accountingApiUrl !== null) {
            $this->accountingApiUrl = $accountingApiUrl;
        }
        if ($taxApiUrl !== null) {
            $this->taxApiUrl = $taxApiUrl;
        }
        if ($blockchainUrl !== null) {
            $this->blockchainUrl = $blockchainUrl;
        }
        if ($apiKey !== null) {
            $this->apiKey = $apiKey;
        }
        if ($timeout !== null) {
            $this->timeout = $timeout;
        }
    }

    public function getAccountingApiUrl(): string
    {
        return $this->accountingApiUrl;
    }

    public function setAccountingApiUrl(string $accountingApiUrl): self
    {
        $this->accountingApiUrl = $accountingApiUrl;
        return $this;
    }

    public function getTaxApiUrl(): string
    {
        return $this->taxApiUrl;
    }

    public function setTaxApiUrl(string $taxApiUrl): self
    {
        $this->taxApiUrl = $taxApiUrl;
        return $this;
    }

    public function getBlockchainUrl(): string
    {
        return $this->blockchainUrl;
    }

    public function setBlockchainUrl(string $blockchainUrl): self
    {
        $this->blockchainUrl = $blockchainUrl;
        return $this;
    }

    public function getApiKey(): string
    {
        return $this->apiKey;
    }

    public function setApiKey(string $apiKey): self
    {
        $this->apiKey = $apiKey;
        return $this;
    }

    public function getTimeout(): int
    {
        return $this->timeout;
    }

    public function setTimeout(int $timeout): self
    {
        $this->timeout = $timeout;
        return $this;
    }
}

File: stacks/backend/JVM/src/main/java/com/chainrice/bridge/config/ChainRiceConfig.java
package com.chainrice.bridge.config;

/**
 * Configuration class for ChainRice client.
 */
public class ChainRiceConfig {
    private String accountingApiUrl = "http://localhost:8080";
    private String taxApiUrl = "http://localhost:8081";
    private String blockchainUrl = "http://localhost:26657";
    private String apiKey = "";
    private int timeout = 30;

    public ChainRiceConfig() {
    }

    public ChainRiceConfig(String accountingApiUrl, String taxApiUrl, String blockchainUrl, String apiKey,
            int timeout) {
        this.accountingApiUrl = accountingApiUrl;
        this.taxApiUrl = taxApiUrl;
        this.blockchainUrl = blockchainUrl;
        this.apiKey = apiKey;
        this.timeout = timeout;
    }

    public String getAccountingApiUrl() {
        return accountingApiUrl;
    }

    public void setAccountingApiUrl(String accountingApiUrl) {
        this.accountingApiUrl = accountingApiUrl;
    }

    public String getTaxApiUrl() {
        return taxApiUrl;
    }

    public void setTaxApiUrl(String taxApiUrl) {
        this.taxApiUrl = taxApiUrl;
    }

    public String getBlockchainUrl() {
        return blockchainUrl;
    }

    public void setBlockchainUrl(String blockchainUrl) {
        this.blockchainUrl = blockchainUrl;
    }

    public String getApiKey() {
        return apiKey;
    }

    public void setApiKey(String apiKey) {
        this.apiKey = apiKey;
    }

    public int getTimeout() {
        return timeout;
    }

    public void setTimeout(int timeout) {
        this.timeout = timeout;
    }
}
